scripts/ingest_nwb_extracellular.py

```diff
@@ -41,24 +41,22 @@
     if utilities.parse_prefix(dob_str.group()) is not None:
         subject_info['date_of_birth'] = utilities.parse_prefix(dob_str.group())
     
-    # source and strain
-    strain_str = re.search('(?<=animalStrain:\s)(.*)', subject_info['description']) # extract the information related to animal strain
-    if strain_str is not None: # if found, search found string to find matched strain in db
-        for s in subject.StrainAlias.fetch():
-            m = re.search(re.escape(s[0]), strain_str.group(), re.I) 
-            if m is not None:
-                subject_info['strain'] = (subject.StrainAlias & {'strain_alias': s[0]}).fetch1('strain')
-                break
-    source_str = re.search('(?<=animalSource:\s)(.*)', subject_info['description'])  # extract the information related to animal strain
-    if source_str is not None:  # if found, search found string to find matched strain in db
-        for s in reference.AnimalSourceAlias.fetch():
-            m = re.search(re.escape(s[0]), source_str.group(), re.I) 
-            if m is not None:
-                subject_info['animal_source'] = (reference.AnimalSourceAlias & {'animal_source_alias': s[0]}).fetch1('animal_source')
-                break
-            
+    # strain
+    strain_str = re.search('(?<=animalStrain:\s)(.*)', subject_info['description']).group() # extract the information related to animal strain
+    strain_dict = {alias: strain for alias, strain in subject.StrainAlias.fetch()}
+    regex_str = ''.join([re.escape(alias) + '|' for alias in strain_dict.keys()])[:-1]
+    strains = [strain_dict[s] for s in re.findall(regex_str, strain_str)]
+    # source
+    source_str = re.search('(?<=animalSource:\s)(.*)', subject_info['description']).group()  # extract the information related to animal strain
+    source_dict = {alias.lower(): source for alias, source in reference.AnimalSourceAlias.fetch()}
```

Comment:
This use of `fetch` is not recommended because it relies on a consistent number and order of attributes in `reference.AnimalSourceAlias` rather than their names.  In DataJoint, attributes are identified by their names rather than their order.
```suggestion
    source_dict = {alias.lower(): source for alias, source in zip(reference.AnimalSourceAlias.fetch('source_alias','source'))}
```